### PR TITLE
ar/adding user list page for admins

### PR DESCRIPTION
**Pull request by angyangie**
**What does this pull request do? Explain your changes. (required)**
- Allows admin user only to see all users created on Livepeer
- Allows admin user to make another user on the list an admin, and to take admin rights away.

_In the future, will need to:(NOT included in this PR)_
- Paginate users
- Add other admin features
- Create side menu to allow for clicking into specific sections of the admin dashboard, like `api-tokens`, `users`, and any others added

**Specific updates (required)**
https://github.com/livepeer/livepeerjs/issues/655

**How did you test each of these updates (required)**
Locally

**Does this pull request close any open issues?**
https://github.com/livepeer/livepeerjs/issues/655

**Screenshots (optional):**

Screen Shot 2020-05-06 at 12 28 54 PM

Screen Shot 2020-05-06 at 12 24 54 PM

Screen Shot 2020-05-06 at 12 25 20 PM

Screen Shot 2020-05-06 at 12 25 32 PM

**Checklist:**
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by wohlner** [comment]
Repeating from discord: Eventually we will have to add some sort of pagination. For now though, I'm ok with it taking a while to load if we have a bunch of users.

File: packages/livepeer.com/www/hooks/use-api.tsx [hunk 1]
```diff
@@ -140,6 +140,35 @@ const makeContext = (state: ApiState, setState) => {
...
+
+      if (res.status !== 201) {
+        return users;
+      }
+      return res;
+    },
+
+    async makeUserAdmin(
+      email,
+      opts = {}
```
Comment (darkdarkdragon) [inline packages/livepeer.com/www/hooks/use-api.tsx]: It is not used, probable should be removed

File: packages/livepeer.com/www/components/TokenTable/index.tsx [hunk 2]
```diff
@@ -140,6 +140,9 @@ export default ({ userId, id }) => {
         </Modal>
       )}
       <p>
+        <p>
+          <strong>Api Tokens</strong>
```
Comment (darkdarkdragon) [inline packages/livepeer.com/www/components/TokenTable/index.tsx]: Maybe 'Streams' title should be made strong too?

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

**Comment by darkdarkdragon** [comment]
@angyangie Maybe we should add button `Remove admin rights`?

**Comment by angyangie** [comment]
@darkdarkdragon true. I'll add a "Remove Admin Rights" button!

File: packages/api/src/schema/schema.yaml [hunk 3]
```diff
@@ -483,6 +500,35 @@ paths:
...
+              type: object
+              $ref: '#/components/schemas/make-admin'
+
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: array
```
Comment (darkdarkdragon) [inline packages/api/src/schema/schema.yaml]: Never worked with these schemas specs, but isn't here specified that return type is array, but actually you're returning object?
```js
res.json({ email: user.email, admin: user.admin })
```

File: packages/livepeer.com/www/components/TokenTable/index.tsx [hunk 4]
```diff
@@ -140,6 +140,9 @@ export default ({ userId, id }) => {
         </Modal>
       )}
       <p>
+        <p>
```
Comment (darkdarkdragon) [inline packages/livepeer.com/www/components/TokenTable/index.tsx]: this gives error in console:
```
Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.
```
Outer <p> should be changed to <div>

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

File: packages/livepeer.com/www/components/UserTable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            </Button>
+          </Flex>
+        </Modal>
+      )}
+      <p>
+        <strong>Users:</strong>
+      </p>
+      <Button
+        variant="secondarySmall"
+        disabled={!selectedUser}
```
Comment (darkdarkdragon) [inline packages/livepeer.com/www/components/UserTable/index.tsx]:
```suggestion
        disabled={!selectedUser || selectedUser.admin}
```

File: packages/livepeer.com/www/components/UserTable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        sx={{ margin: 2, mb: 4 }}
+        onClick={() => selectedUser && setAdminModal(true)}
+      >
+        Make User Admin
+      </Button>
+      <Button
+        variant="secondarySmall"
+        disabled={!selectedUser}
```
Comment (darkdarkdragon) [inline packages/livepeer.com/www/components/UserTable/index.tsx]:
```suggestion
        disabled={!selectedUser || !selectedUser.admin}
```

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

File: packages/api/src/schema/schema.yaml (hunk 3 above)
Comment (angyangie) [reply packages/api/src/schema/schema.yaml]: Ah yes, I missed that one!

File: packages/livepeer.com/www/components/TokenTable/index.tsx (hunk 4 above)
Comment (angyangie) [reply packages/livepeer.com/www/components/TokenTable/index.tsx]: I added a couple of `Boxes` instead

**Review by darkdarkdragon — APPROVED** [review]
LGTM!

**Comment by wohlner** [comment]
LGTM

Eventually, we will want to be able to sort by admin status, but that can be another ticket / PR. I'll document it in zenhub.

**Pull request merged**